### DISCUSSION TITLE
Swupdate on xavier nx branch kirkstoner32.7

### DISCUSSION THREAD

**Discussion by yrodriguez31** [OE4T/meta-tegra#1632] [Q&A]
Hello ,

I have issues integrating swupdate on my distro.

I tried first of all to start from [tegra-test-distro](https://github.com/madisongh/tegra-test-distro/tree/master)

And i follow the instructions of this [topic](https://github.com/OE4T/meta-tegra/discussions/842)

I tried to integrate evrything but when i bbapend swupdate with this  [files ](https://github.com/madisongh/tegra-test-distro/tree/master/layers/meta-tegra-support/dynamic-layers/meta-swupdate/recipes-support/swupdate) swupdate doesn't build because of the patch  [0001-Add-support-for-custom-bootloader.patch](https://github.com/madisongh/tegra-test-distro/blob/master/layers/meta-tegra-support/dynamic-layers/meta-swupdate/recipes-support/swupdate/swupdate/0001-Add-support-for-custom-bootloader.patch) .

So in my distro i have all the patch on uboot from tegra demo distro and this [patches](https://github.com/madisongh/tegra-test-distro/tree/master/layers/meta-testdistro/recipes-support/swupdate) for swupdate. 

But i have this error when i tried to update my jetson 

```

SWUpdate v2023.12

Licensed under GPLv2. See source distribution for detailed copyright notices.

[INFO ] : SWUPDATE running :  [main] : Using default bootloader interface: uboot
[INFO ] : SWUPDATE running :  [lua_handlers_init] : External Lua handler(s) not found.
[INFO ] : SWUPDATE running :  [main] : Running on jetson-xavier-nx-orphie Revision 1.0
[INFO ] : SWUPDATE running :  [print_registered_handlers] : Registered handlers:
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	dummy
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	archive
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	tar
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	flash
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	diskformat
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	lua
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	raw
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	rawfile
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	shellscript
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	preinstall
[INFO ] : SWUPDATE running :  [print_registered_handlers] : 	postinstall
[INFO ] : SWUPDATE running :  [main] : software set: system mode: slot_a
[INFO ] : SWUPDATE started :  Software Update started !
[ERROR] : SWUPDATE failed [0] ERROR : Cannot initialize environment from /etc/fw_env.config
[ERROR] : SWUPDATE failed [0] ERROR : Cannot initialize environment from /etc/fw_env.config
[WARN ] : SWUPDATE running :  [update_transaction_state] : Cannot persistently store in_progress update state.
```
Do you have an idea of what is missing on my distro ? 

Regards

**Comment by madisongh** [comment]
U-Boot is not used on the Xavier hardware.  The reason for the custom-bootloader patch that you dropped was to provide a way to interface swupdate with a bootloader it doesn't have support for. I don't know if swupdate has changed since I wrote that patch, to provide a better way to do this, but you'll need something other than U-Boot support.

**Reply by yrodriguez31** [reply]
Ok thanks for your Replay. So the fw_env file is only for uboot ?

**Reply by madisongh** [reply]
Yes

**Reply by yrodriguez31** [reply]
Ok thanks
